packages/mobile/src/screens/chat-screen/ChatUserListScreen.tsx

```diff
@@ -182,34 +220,24 @@ export const ChatUserListScreen = (props: ChatUserListScreenProps) => {
 
   useDebounce(
     () => {
-      dispatch(searchUsers({ query }))
-      setHasQuery(!!query)
-    },
-    debounceMs,
-    [query, setHasQuery, dispatch]
-  )
-
-  const handleChange = useCallback(
-    (text: string) => {
-      setQuery(text)
+      setQuery(inputValue)
```

Comment:
interesting.. so you're debouncing setting the state var instead of the dispatch call. i think i see why this is better but if you have a sec to explain would appreciate :)

Reply:
It's functionally not much different than what was here already(one state value is directly tied to the input and updates immediately on change, and that value is synchronized to the rest of the state on a debounce). But this pattern feels easier to follow because all of the code depending on `query` can respond when its value changes instead of some things tracking `hasQuery` and some things tracking the value of the query itself.

So I think it was mostly that I didn't like two different state variables when one would do fine?